app/controllers/spree/orders_controller.rb

```diff
@@ -180,10 +180,10 @@ def check_at_least_one_line_item
       items = params[:order][:line_items_attributes]
         &.select{ |_k, attrs| attrs["quantity"].to_i > 0 }
 
-      if items.empty?
-        flash[:error] = I18n.t(:orders_cannot_remove_the_final_item)
-        redirect_to main_app.order_path(order_to_update)
-      end
+      return unless items.empty?
+
+      flash[:error] = I18n.t(:orders_cannot_remove_the_final_item)
+      redirect_to main_app.order_path(order_to_update)
     end
```

Comment:
The guard-clause rule is mostly good but there are cases like this where it's no appropriate, in my opinion. Guard clauses are good to return early on error cases while keeping the main part of the method at the end without indentation. But in this case, the main part of the method's code already ran, removing items, and the changed part is an error case. The previous `if` statement made that more clear.

To be fair, the redirect should probably be in the controller action and not in this method. But that's out of scope here. I'm accepting this correction because there's no easy way to resolve except disabling the rule here which is clutter as well.

But I'm mentioning it because there may be other cases where the cop isn't providing a better solution.